Controller/SitemapController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2014 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\SeoBundle\Controller;
+
+use Symfony\Cmf\Bundle\SeoBundle\Model\UrlInformation;
+use Symfony\Cmf\Bundle\SeoBundle\Sitemap\UrlInformationProviderInterface;
+use Symfony\Component\Config\Definition\Exception\InvalidConfigurationException;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Templating\EngineInterface;
+
+/**
+ * Controller to handle requests for sitemap.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@gmx.de>
+ */
+class SitemapController
+{
+    /**
+     * @var UrlInformationProviderInterface
+*/
+    private $urlProvider;
+    /**
+     * @var EngineInterface
+     */
+    private $templating;
+
+    /**
+     * @var array
+     */
+    private $templates;
+
+    /**
+     * Inject the templates as an hashmap like:
+     *
+     * array('<format>' => 'Bundle:Domain:template.format.twig')
+     *
+     * Valid formats are html, json and xml.
+     *
+     * @param UrlInformationProviderInterface $provider
+     * @param EngineInterface $templating
+     * @param array $templates
+     */
+    public function __construct(
+        UrlInformationProviderInterface $provider,
+        EngineInterface $templating,
+        array $templates
+    ) {
+        $this->urlProvider = $provider;
+        $this->templating = $templating;
+        $this->templates = $templates;
+    }
+
+    /**
+     * @param string $_format The format of the sitemap. Supported values are html|xml|json
```

Comment:
Remove the thing about supported values, now that we allow to inject any templates.